core/src/test/java/feast/core/service/JobServiceTest.java

```diff
@@ -345,6 +340,12 @@ public void testStopUnsupportedError() {
     this.job.setStatus(prevStatus);
   }
 
+  @Test(expected = NoSuchElementException.class)
+  public void testStopJobForUnknownId() {
+    var request = StopIngestionJobRequest.newBuilder().setId("bogusJobId").build();
+    jobService.stopJob(request);
```

Comment:
This didn't fail before the fix since it was the same exception type anyway. I don't think there's an easy way to assert on a thrown exception's message in JUnit 4.

Reply:
I think Gojek folks tried/wanted to before—the Java SDK already uses it—but there was some reason it could not or should not be done? IIRC either @davidheryanto or @zhilingc mentioned it to me somewhere awhile back.